File: katas/6 kyu/data-reverse/solution.py
# data-reverse
import codewars_test as test
# -------------------------------- MY SOLUTION ------------------------------- #
def data_reverse(data):
  bytes = [data[x:x+8] for x in range(len(data)) if x%8 == 0][::-1]
  return [i for x in bytes for i in x]
# ------------------------------ CLEVER SOLUTION ----------------------------- #
def data_reverse(data):
  res = []
  
  for i in range(len(data)-8, -1, -8):
    res.extend(data[i:i+8])
  
  return res
# ----------------------------------- TEST ----------------------------------- #
data1 = [1,1,1,1,1,1,1,1,0,0,0,0,0,0,0,0,0,0,0,0,1,1,1,1,1,0,1,0,1,0,1,0]
data2 = [1,0,1,0,1,0,1,0,0,0,0,0,1,1,1,1,0,0,0,0,0,0,0,0,1,1,1,1,1,1,1,1]
test.assert_equals(data_reverse(data1),data2)

data3 = [0,0,1,1,0,1,1,0,0,0,1,0,1,0,0,1]
data4 = [0,0,1,0,1,0,0,1,0,0,1,1,0,1,1,0]
test.assert_equals(data_reverse(data3),data4)

File: katas/6 kyu/data-reverse/solution.js
// data-reverse
/* ------------------------------- MY SOLUTION ------------------------------ */
function dataReverse(data) {
  out = []
  for (let i = 0; i < data.length; i = i+8) {
    const x = data.slice(i , i+8);
    out.push(x)
  }
  out.reverse()
  return out.reduce((p , n) => p.concat(n) , [])
}
/* ----------------------------- CLEVER SOLUTION ---------------------------- */
/* ---------------------------------- TEST ---------------------------------- */
// const chai = require("chai");
// const Test = chai.assert.strictEqual; // replace Test.assertEqual with Test
// chai.config.truncateThreshold=0;
const chai = require("chai");
const assert = chai.assert;
chai.config.truncateThreshold=0;

describe("Data Reverse", function(){
  it("Example Tests", function(){
    const data1 = [1,1,1,1,1,1,1,1,0,0,0,0,0,0,0,0,0,0,0,0,1,1,1,1,1,0,1,0,1,0,1,0]
    const data2 = [1,0,1,0,1,0,1,0,0,0,0,0,1,1,1,1,0,0,0,0,0,0,0,0,1,1,1,1,1,1,1,1]
    assert.deepEqual(dataReverse(data1),data2)
    const data3 = [0,0,1,1,0,1,1,0,0,0,1,0,1,0,0,1]
    const data4 = [0,0,1,0,1,0,0,1,0,0,1,1,0,1,1,0]
    assert.deepEqual(dataReverse(data3),data4)
  });
});